### ISSUE TITLE
ReCAPTCHA settings null after using AddRecaptcha options

### ISSUE THREAD

**Issue by samuelgrant** [TimothyMeadows/reCAPTCHA.AspNetCore#62]
Hi..

Your fix for #60 has worked so far as it stops the application from crashing on runtime, however it seems the settings set using the options method do not persist when they are injected into a controller... 

**Read me example being used:**
```cs
// Or configure recaptcha via options
services.AddRecaptcha(options =>
{
    options.SecretKey = "Your secret key";
    options.SiteKey = "Your site key";
});
```

![image](https://user-images.githubusercontent.com/15701407/85644346-c0b65a00-b6ea-11ea-99e5-fc667d16720d.png)

I noticed this issue when trying to validate responses. The Validate(code) method was returning a new response object with all fields set to the default value. It may be worth throwing some sort of exception if a validation is attempted without a secret key..

![image](https://user-images.githubusercontent.com/15701407/85644481-299dd200-b6eb-11ea-9355-afa23ccc6e65.png)

**Comment by TimothyMeadows** [comment]
Interesting, will look into this.

**Comment by TimothyMeadows** [comment]
![image](https://user-images.githubusercontent.com/4072037/85730233-58e21e00-b6ae-11ea-9141-0ec77ccd8d93.png)

Seems to be just fine? Where are you getting the settings object from?

**Comment by TimothyMeadows** [comment]
Seems fine with @inject too 
![image](https://user-images.githubusercontent.com/4072037/85730654-afe7f300-b6ae-11ea-9833-10de5e1238b7.png)

**Comment by TimothyMeadows** [comment]
Looks like this is a result of not using services.Configure in the AddRecaptcha methods that were recently added by @mo-esmp . Rather services.AddSingleton is called for the object which is injecting settings as RecatpchaSettings and not IOptions<RecaptchaSettings>.

In the future to follow proper standard you should always call services.Configure with the "options". This method supports Action<RepcatchaSettings> by default on it's own.

**Comment by TimothyMeadows** [comment]
This is now fixed in 3.0.5.

**Comment by mo-esmp** [comment]
Yes, I removed `IOptions` but you rolled back code to `IOptions`. I added some integration tests to validate captcha registration and I think it worth to run tests after any change.
